### PR TITLE
drivers: pwm: Remove all deprecated APIs.

### PR DESCRIPTION
**Pull request by AdithyaBaglody**
JIRA: ZEP-2018

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

### CODE REVIEW

**Comment by AdithyaBaglody** [comment]
DO NOT MERGE until #943 is merged.

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by andrewboie** [comment]
#943 now merged, this can go in

**Comment by nashif** [comment]
> #943 now merged, this can go in

merged into ARM branch, still not in master :(

**Pull request merged**
